lib/vmdb/plugins.rb

```diff
@@ -80,6 +86,48 @@ def register_models
 
     private
 
+    # Determine the version of the specified engine
+    #
+    # If the gem is
+    # - git based, pointing to a branch:   <branch>@<sha>
+    # - git based, pointing to a tag:      <tag>@<sha>
+    # - git based, pointing to a sha:      <sha>
+    # - path based, with git, on a branch: <branch>@<sha>
+    # - path based, with git, on a tag:    <tag>@<sha>
+    # - path based, with git, on a sha:    <sha>
+    # - path based, without git:           nil
+    # - a real gem:                        <gem_version>
+    def version(engine)
+      spec = bundler_specs_by_path[engine.root.to_s]
+
+      case spec.source
+      when Bundler::Source::Git
+        [
+          spec.source.branch || spec.source.options["tag"],
+          spec.source.revision.presence[0, 8]
+        ].compact.join("@").presence
+      when Bundler::Source::Path
+        if engine.root.join(".git").exist?
+          branch = sha = nil
+          Dir.chdir(engine.root) do
+            branch   = `git rev-parse --abbrev-ref HEAD 2>/dev/null`.strip.presence
+            branch   = nil if branch == "HEAD"
```

Comment:
I don't yet have branch to do the conditional that way

I am throwing it away on purpose because I don't want HEAD as the branch name.  These 3 lines fetch the "name" of the branch, however if the branch is "HEAD", then we are actually in a detached HEAD state, so we need to check if we are on a tag instead.  I set it to nil because we don't want HEAD to be our branch name, and if the tag is nil it will just stay nil.

Reply:
Oh right... clearly I was drunk when I made that comment.  Carry on.